code/modules/food_and_drinks/machinery/microwave.dm

```diff
@@ -381,12 +381,20 @@
 	if(operating)
 		return NONE
 
-	if (item.item_flags & ABSTRACT)
+	if(item.item_flags & ABSTRACT)
+		return NONE
+
+	if(dirty >= MAX_MICROWAVE_DIRTINESS) // The microwave is all dirty so can't be used!
+		if(IS_EDIBLE(item))
```

Comment:
Technically, you can microwave all items, not just edibles. These checks are going to obscure the balloon alerts and make players bash the microwave when they didn't mean to. Overall this `item_interaction` is quite inconsistent. It does check for combat mode, but only by the end of the proc.

Reply:
Yes the problem is right now you literally can't attack or use tools on a microwave if it's dirty even if it's otherwise sensible to, this whole proc is a mess but I didn't want to have to rewrite the entire thing since it probably needs extra special attention

Reply:
Eugh, ok fine...